scripts/run_portserver_test.py

```diff
@@ -0,0 +1,502 @@
+# coding: utf-8
```

Comment:
My overall thought on `run_portserver_test.py` is that you might be over-complicating things by testing all the helper functions individually and then mocking them. The portserver isn't really very complicated--it just hands out port numbers upon request making sure no two clients ever have any given port number at the same time. I think your tests will be simpler (and you won't have to deal with so many private attributes) if you test this interface from the client's perspective instead

Reply:
Thanks @U8NWXD.
I have simplified the tests now. I converted some of the private methods to public because getting all of their branches covered was difficult as some of them had their wrapper methods declared private too. PTAl. Thanks!